spec/models/azure_engine_spec.rb

```diff
@@ -0,0 +1,181 @@
+      def name
+        'dummy'
+      end
+    end
+    class DummyBlobService
+      def create_block_blob(container, key, content)
+        nil
+      end
+      def list_blobs(container, options = {})
+        result = []
+        result.push(Azure::Blob::DummyBlobObject.new)
+        result
+      end
+      def get_blob(container, key)
+        return nil, 'hoge'
+      end
+      def delete_blob(container, key)
+        nil
+      end
+      def generate_uri(uri, options = {})
+        'http://www.example.com'
+      end
+    end
+  end
+
+  module Contrib
+    module Auth
+      class DummySharedAccessSignature
+        def sign
+          'https://signed.example.com'
+        end
+      end
+    end
+  end
+end
+
+describe 'AzureEngine' do
+  before do
+    AzureEngine.configure do |config|
+      config.bucket_name = 'my-bucket'
+      config.image_bucket_name = 'my-image-bucket'
+      config.cdn_base_url = ''
+      config.queue_name = 'my-queue'
+      config.azure_storage_access_key = 'azure_storage_access_key'
+      config.azure_storage_account_name = 'azure_storage_account_name'
+    end
+  end
+
+  describe 'resource_endpoint' do
+    it 'return URL that includes image bucket name' do
+      expect(AzureEngine.resource_endpoint).to eq('https://azure_storage_account_name.blob.core.windows.net/my-image-bucket')
+    end
+    it 'return CDN URL' do
+      AzureEngine.configure do |config|
+        config.cdn_base_url = 'https://sushi.example.com'
+      end
+      expect(AzureEngine.resource_endpoint).to eq('https://sushi.example.com/my-image-bucket')
+    end
+  end
+
+  describe 'upload_endpoint' do
+    it 'return URL that includes bucket name' do
+      expect(AzureEngine.upload_endpoint).to eq('https://azure_storage_account_name.blob.core.windows.net/my-bucket')
+    end
+  end
+
+  describe 'queue_endpoint' do
+    it 'return URL that specifies queue endpoint' do
+      expect(AzureEngine.queue_endpoint).to eq('https://azure_storage_account_name.queue.core.windows.net/')
+    end
+  end
+
+  describe 'Blob Queue' do
+    before do
+      allow_any_instance_of(Azure::ClientServices).to receive(:queues).and_return(Azure::Queue::DummyQueueService.new)
+    end
+
+    it 'succeeds to send message' do
+      expect(AzureEngine.send_message('hoge')).to eq(nil)
+    end
+
+    it 'succeeds to receive message' do
+      expect(AzureEngine.receive_message(10)).to eq([])
+    end
+
+    it 'succeeds to return correct status of the existence of a message' do
+      expect(AzureEngine.message_exist?([])).to eq(false)
+      expect(AzureEngine.message_exist?([Azure::Entity::Queue::DummyMessage.new])).to eq(true)
+    end
+
+    it 'succeeds to delete message' do
+      expect(AzureEngine.delete_message(Azure::Entity::Queue::DummyMessage.new)).to eq(nil)
+    end
+
+    it 'succeeds to delete messages all at once' do
+      expect(AzureEngine.batch_delete([Azure::Entity::Queue::DummyMessage.new]).class.name).to eq("Array")
```

Comment:
__[rubocop]__
* Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)